bindings/java/devdoc/java_out_of_process_hld.md

```diff
@@ -0,0 +1,122 @@
+Java Out of Process Module API
```

Comment:
the java module receives the args as String so there is no parseJsonConfig or freeConfig.  The module can parse the args internally.